clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -867,6 +869,7 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
             this.subscriptions.subscribe(pattern, listener);
             this.metadata.needMetadataForAllTopics(true);
             this.metadata.requestUpdate();
+            this.coordinator.updatePatternSubscription(metadata.fetch());
```

Comment:
@becketqin I think we're covered because we don't change the assignment until the rebalance (at which point the autocommit is triggered).